### PR TITLE
docs itk org

### PR DESCRIPTION
**Pull request by thewtex**
This adds Myst (markdown) Sphinx documentation. This is to be rendered on
ReadTheDocs and deployed on docs.itk.org. A start for issue #4063.
Changes and more content are expected in the future! In particular, the
old wiki content, Software Guide content, and Tutorial. It
uses the clean and elegant Furo theme, which works with Myst and is also
used by docs.vtk.org and wasm.itk.org.
      
This incorporates the markdown files in the Documentation/ directory. It
renders a clean build without warnings. A few changes were required for
to avoid warnings, e.g. header specifications and cross-link
specifications.
      
Symlinks were created at the top level for CODE_OF_CONDUCT.md and
CONTRIBUTING.md, but these should be replace by real files with links to
the rendered content once published for GitHub access.

### CODE REVIEW

**Comment by thewtex** [comment]
This is a start for #4063. It is a starting point -- there are many updates and changes and additions to be made, but I would like to get the current content deployed to use an updated version Download page for ITK 5.4 RC 2.

For an ITK community hackathon in November, I think we should work on migrating the old Wiki content.

Eventually the ITK Software Guide, tutorial. The Sphinx Examples we could keep in their own repository, since they are so large, but we may want to migrate to this theme / Markdown.

**Review by jhlegarreta — COMMENTED** [review]
Wonderful :rocket:. Thanks for the effort Matt. Having all SWG and Examples rendered from a single site would certainly offer a better user experience.

Correct me if I'm wrong but the LaTeX code in SWG wouldn't need to be translated to Markdown, right?

kw ghostflow is reporting some trailing whitespaces and missing newlines at EOL (:+1: great that we got them identified incidentally).

**Comment by jhlegarreta** [comment]
BTW, I see a `readthedocs` config file: there is the possibility of getting PR build previews for `readthedocs`. @thewtex let me know if you run into trouble getting those builds working. It took me a while to have them in another repository I contribute to. The caveat is that we will not have them for PRs (e.g. this one) that were opened prior to making the config work in the `readthedocs` settings.

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -1,434 +0,0 @@
-ITK Community Participation Guidelines
```
Comment (jcfr) [inline CODE_OF_CONDUCT.md:1]: Instead, you could keep the file at the root, and consider `Documentation/docs/contributing/code_of_conduct.md` like this:

<pre>
```{include} ../../CODE_OF_CONDUCT.md
.md
```
</pre>

File: Documentation/docs/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+For example in Linux / macOS:
+
+```
+python -m venv env
+source env/bin/activate
+pip install -r requirements.txt
+```
+
+Use `make html` in this directory to build the documentation.
+Open `_build/html/index.html` in your browser to inspect the result.
```
Comment (jcfr) [inline Documentation/docs/README.md:18]: You may want to include a section like this one:
https://github.com/slicer/slicer.org#regenerating-favicons

File: Documentation/docs/conf.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+extensions = [
+    'sphinx.ext.napoleon',
+    'myst_parser',
+    'sphinx.ext.intersphinx',
+    'sphinx_copybutton',
+    'sphinxext.opengraph',
+    'sphinx_design',
+]
+
+myst_enable_extensions = ["colon_fence", "fieldlist"]
```
Comment (jcfr) [inline Documentation/docs/conf.py:24]:
```suggestion
myst_enable_extensions = [
    "colon_fence",
    "dollarmath",  # Support syntax for inline and block math using `$...$` and `$$...$$`
                   # (see https://myst-parser.readthedocs.io/en/latest/syntax/optional.html#dollar-delimited-math)
    "fieldlist",
    "linkify",  # convert bare links to hyperlinks
]
```

File: Documentation/docs/conf.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+
+intersphinx_mapping = {
+    "python": ("https://docs.python.org/3/", None),
+    "numpy": ("https://numpy.org/doc/stable", None),
+}
+
+html_theme = 'furo'
+html_static_path = ['_static']
+# html_logo = "_static/itk-logo.svg"
```
Comment (jcfr) [inline Documentation/docs/conf.py:37]:
```suggestion
```

File: Documentation/docs/contributing/enforcement.md [hunk 5]
```diff
@@ -70,8 +70,9 @@ In this document, the term "the originator" refers to the person(s) that have
 violated this Code of Conduct and originated an incident; the term "the
 affected party(ies)" refers to the person(s) that have suffered the attack of "the
 originator" under the forms described in the
-[Unacceptable Behavior](CODE_OF_CONDUCT.md#unacceptable-behavior) section.
+[Unacceptable Behavior](code_of_conduct.md) section.
```
Comment (jcfr) [inline Documentation/docs/contributing/enforcement.md:73]: Did you remove the anchor on purposed ?

File: Documentation/docs/index.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
+# Welcome to ITK's documentation!
+
+The Insight Toolkit (ITK) is an open-source, cross-platform toolkit for N-dimensional scientific image processing, segmentation, and registration.
+Our documentation is a resource for
+both new and seasoned ITK community members and includes tutorials, examples, and best
+practices to help you unlock the full power of ITK's advanced algorithms.
```
Comment (jcfr) [inline Documentation/docs/index.md:6]:
```suggestion
Our documentation is a resource for both new and seasoned ITK community members and includes tutorials, examples, and best practices to help you
unlock the full power of ITK's advanced algorithms.

File: Documentation/docs/requirements.txt [hunk 7]
```diff
@@ -0,0 +1,6 @@
+sphinx>=7.2.6
+furo
+myst-parser
+sphinx-copybutton
+sphinxext-opengraph
+sphinx-design
```
Comment (jcfr) [inline Documentation/docs/requirements.txt:6]:
```suggestion
furo
myst-parser
sphinx>=7.2.6
sphinx-copybutton
sphinx-design
sphinxext-opengraph
```

**Review by jcfr — COMMENTED** [review]
:tada:  This is a really nice improvement :100:

**Review by phcerdan — APPROVED** [review]
:tada:

**Review by dzenanz — COMMENTED** [review]
Looks good at a glance.

**Comment by thewtex** [comment]
> Correct me if I'm wrong but the LaTeX code in SWG wouldn't need to be translated to Markdown, right?

> My memories are a bit vague now, but I remember the VTK site being able to process LaTeX data.

For the VTK site, they migrated to Markdown. With a migration to markdown, we can also add website interactivity.

> kw ghostflow is reporting some trailing whitespaces and missing newlines at EOL (👍 great that we got them identified incidentally).

There are unrelated to this change. I will address the trailing white space and missing newlines. I think the other "issues", mostly related to how the original release notes were written, can be ignored.

> BTW, I see a readthedocs config file: there is the possibility of getting PR build previews for readthedocs.

Yes, let's get PR build previews, and set that up after this has been integrated.

File: Documentation/docs/contributing/enforcement.md (hunk 5 above)
Comment (thewtex) [reply Documentation/docs/contributing/enforcement.md:73]: Yes -- cross-document references were not supported by Myst.

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (thewtex) [reply CODE_OF_CONDUCT.md:1]: This did not work as desired, at least for me.

File: Documentation/docs/README.md (hunk 2 above)
Comment (thewtex) [reply Documentation/docs/README.md:18]: I added a simplified version in *_static/icon/README.md*.

**Review by jhlegarreta — APPROVED** [review]

File: Documentation/docs/conf.py (hunk 3 above)
Comment (thewtex) [reply Documentation/docs/conf.py:24]: :+1: added

**Comment by thewtex** [comment]
> Having all SWG and Examples rendered from a single site would certainly offer a better user experience.

Yes, I agree, but I think we have to find the right balance with documentation development build time. Currently, this takes about 20 seconds to build. Adding the Software Guide and Wiki content, it could still be reasonable. I think the Examples are already take too long to build.

**Comment by thewtex** [comment]
ghostflow checks errors are false positives:

> Documentation/docs/make.bat:1: trailing whitespace.

There are intentional carriage returns since this is a Windows batch file.

**Comment by jhlegarreta** [comment]
The docs look beautiful :100::
https://docs.itk.org/en/latest/

**Comment by jhlegarreta** [comment]
BTW, the link should appear in https://itk.org/ and the [README file](https://github.com/InsightSoftwareConsortium/ITK#links).

**Pull request merged**